### PR TITLE
ggml : generalize GGML_OP_CONCAT

### PR DESCRIPTION
**Pull request by ggerganov**
Support concat across all dimensions:

https://pytorch.org/docs/stable/generated/torch.cat.html#torch.cat

### CODE REVIEW

**Comment by ggerganov** [comment]
The CUDA implementation probably should be improved - I tried to keep the original logic, but I think it would be better to have a single kernel accepting `dim` as argument

**Review by slaren — APPROVED** [review]

**Comment by ggerganov** [comment]
I realized there was a bug in yesterday's CPU/Metal implementation. Should be good now
